src/Files.App/Filesystem/RecentItemsManager.cs

```diff
@@ -109,18 +111,32 @@ public async Task<List<RecentItem>> ListRecentFilesAsync()
         /// <summary>
         /// Enumerate recently accessed folders via `Windows\Recent`.
         /// </summary>
-        public async Task<List<RecentItem>> ListRecentFoldersAsync()
+        public List<RecentItem> ListRecentFolders()
         {
-            List<RecentItem> linkItems = null;
+            var recentItems = new List<RecentItem>();
+            var excludeMask = FileAttributes.Hidden;
+            var linkFilePaths = Directory.EnumerateFiles(CommonPaths.RecentItemsPath).Where(f => (new FileInfo(f).Attributes & excludeMask) == 0);
 
-            var (status, response) = await SendRecentItemsActionForResponse("EnumerateFolders");
-            if (status == AppServiceResponseStatus.Success && response.ContainsKey("EnumerateFolders"))
+            foreach (var linkFilePath in linkFilePaths)
             {
-                linkItems = JsonConvert.DeserializeObject<List<ShellLinkItem>>((string)response["EnumerateFolders"])
-                                       .Select(link => new RecentItem(link)).ToList();
+                using var link = new ShellLink(linkFilePath, LinkResolution.NoUIWithMsgPump, null, TimeSpan.FromMilliseconds(100));
+
+                try
+                {
+                    if (!string.IsNullOrEmpty(link.TargetPath) && link.Target.IsFolder)
+                    {
+                        var shellLinkItem = ShellFolderExtensions.GetShellLinkItem(link);
+                        recentItems.Add(new RecentItem(shellLinkItem));
+                    }
+                }
+                catch (FileNotFoundException)
+                {
+                    // occurs when shortcut or shortcut target is deleted and accessed (link.Target)
+                    // consequently, we shouldn't include the item as a recent item
```

Comment:
Can you log this as well? We should include a custom message for this one to indicate it's expected when the target was removed. Having this in the log file will help us in the event this fails for a different reason.

Reply:
This only catches `FileNotFoundException`. If it failed for other reasons, I think it will crash.

Reply:
Done.